characterization-project/src/main/java/dc/Robot.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * This is a very simple robot program that can be used to send telemetry to
+ * the data_logger script to characterize your drivetrain. If you wish to use
+ * your actual robot code, you only need to implement the simple logic in the
+ * autonomousPeriodic function and change the NetworkTables update rate
+ */
+
+package dc;
+
+import java.util.function.Supplier;
+
+import com.revrobotics.CANEncoder;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.TimedRobot;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+public class Robot extends TimedRobot {
+
+  static private double GEARING = 1.375;
+  static private int PIDIDX = 0;
+
+  Joystick stick;
+
+  CANSparkMax master;
+  CANEncoder encoder;
+
+  Supplier<Double> encoderPosition;
+  Supplier<Double> encoderRate;
+
+  NetworkTableEntry autoSpeedEntry =
+      NetworkTableInstance.getDefault().getEntry("/robot/autospeed");
+  NetworkTableEntry telemetryEntry =
+      NetworkTableInstance.getDefault().getEntry("/robot/telemetry");
+
+  double priorAutospeed = 0;
+  Number[] numberArray = new Number[6];
+
+  @Override
+  public void robotInit() {
+    if (!isReal()) SmartDashboard.putData(new SimEnabler());
+
+    stick = new Joystick(0);
+
+    master = new CANSparkMax(7, MotorType.kBrushless);
+    master.setInverted(false);
+    master.setIdleMode(IdleMode.kBrake);
+
+    encoder = master.getEncoder();
+
+    CANSparkMax slave0 = new CANSparkMax(14, MotorType.kBrushless);
+    slave0.follow(master, true);
+    slave0.setIdleMode(IdleMode.kBrake);
+
+    //
+    // Configure encoder related functions -- getDistance and getrate should
+    // return units and units/s
+    //
+    
+    double encoderConstant = (1 / GEARING) * 1;
+
+    encoderPosition = ()
+        -> encoder.getPosition() * encoderConstant;
+    encoderRate = ()
+        -> encoder.getVelocity() * encoderConstant / 60.;
+
```

Comment:
I suggest using an average of both encoders, or collecting each encoder's data

Reply:
This is actually autogenerated via the frc-characterization tool that WPILib offers. For tomorrow I'd like to use the stock generated files without modification to see what output we get.